Matchismo/Matchismo/View/PlayingCardView.mm

```diff
@@ -0,0 +1,213 @@
+                      [NSString stringWithFormat:@"%@\n%@", [self rankAsString], self.suit]
+                                                    attributes:attributes];
+  CGRect textBounds;
+  textBounds.origin = CGPointMake([self cornerOffset], [self cornerOffset]);
+  textBounds.size = [cornerText size];
+  [cornerText drawInRect:textBounds];
+  
+  const auto context = UIGraphicsGetCurrentContext();
+  CGContextTranslateCTM(context, self.bounds.size.width, self.bounds.size.height);
+  CGContextRotateCTM(context, M_PI);
+  [cornerText drawInRect:textBounds];
+  
+}
+
+- (void)drawPips {
+  if ((self.rank == 1)
+      || (self.rank == 3)
+      || (self.rank == 5)
+      || (self.rank == 9)) {
+    [self drawPipsWithHorizontalOffset:0
+                        verticalOffset:0
+                    mirroredVertically:NO];
+  }
+  if ((self.rank == 6)
+      || (self.rank == 7)
+      || (self.rank == 8)) {
+    [self drawPipsWithHorizontalOffset:PIP_HOFFSET_PERCENTAGE
+                        verticalOffset:0
+                    mirroredVertically:NO];
+  }
+  if ((self.rank == 2)
+      || (self.rank == 3)
+      || (self.rank == 7)
+      || (self.rank == 8)
+      || (self.rank == 10)) {
+    [self drawPipsWithHorizontalOffset:0
+                        verticalOffset:PIP_VOFFSET2_PERCENTAGE
+                    mirroredVertically:(self.rank != 7)];
+  }
+  if ((self.rank == 4)
+      || (self.rank == 5)
+      || (self.rank == 6)
+      || (self.rank == 7)
+      || (self.rank == 8)
+      || (self.rank == 9)
+      || (self.rank == 10)) {
+    [self drawPipsWithHorizontalOffset:PIP_HOFFSET_PERCENTAGE
+                        verticalOffset:PIP_VOFFSET3_PERCENTAGE
+                    mirroredVertically:YES];
+  }
+  if ((self.rank == 9) || (self.rank == 10)) {
+    [self drawPipsWithHorizontalOffset:PIP_HOFFSET_PERCENTAGE
+                        verticalOffset:PIP_VOFFSET1_PERCENTAGE
+                    mirroredVertically:YES];
+  }
+}
+
+- (void)drawPipsWithHorizontalOffset:(CGFloat)hOffset
+                      verticalOffset:(CGFloat)vOffset
+                          upsideDown:(BOOL)upsideDown {
+  if (upsideDown) {
+    [self pushContextAndRotateUpsideDown];
+  }
+  const auto middle = CGPointMake(self.bounds.size.width / 2, self.bounds.size.height / 2);
+  auto pipFont = [UIFont preferredFontForTextStyle:UIFontTextStyleBody];
+  pipFont = [pipFont fontWithSize:
+                pipFont.pointSize * self.bounds.size.width * PIP_FONT_SCALE_FACTOR];
+  const auto attributedSuit = [[NSAttributedString alloc]
+                                 initWithString:self.suit
+                                     attributes:@{NSFontAttributeName : pipFont,
+                                                  NSForegroundColorAttributeName : [self textColor]
+                                                }];
+  const auto pipSize = attributedSuit.size;
+  auto pipOrigin = CGPointMake(middle.x-pipSize.width/2-hOffset*self.bounds.size.width,
```

Comment:
All this stuff here is quite hard to read and make sense out of.
Would probably benefit from separating it out a bit, using more named variables, potentially breaking down this large method... and so on.